File: algorithms/c++/strings.cpp
#include <iostream>

bool isPalindrome(std::string word) {
    transform(word.begin(), word.end(), word.begin(), ::toupper);
    int left = 0;
    int right = word.size() - 1;
    while (word[left] == word[right] && left < right) {
        left++;
        right--;
    }

    return word[left] == word[right];
}

int main () {
    std::string word;
    std::cout << "Enter a word to check if it is a palindrome: ";
    std::cin >> word;

    bool wordIsPalindrome = isPalindrome(word);
    if (wordIsPalindrome) {
        std::cout << word << " is a palindrome!" << "\n";
    } else {
        std::cout << word << " is not a palindrome." << "\n";
    }

    return 0;
}

File: algorithms/python/strings.py
def isPalindrome(word):
    left = 0
    right = len(word) - 1

    while word[left] == word[right] and left < right:
        left += 1
        right -= 1

    return word[left] == word[right]


if __name__ == "__main__":
    word = input("Enter a word to check if it is a palindrome: ")
    wordIsPalindrome = isPalindrome("".join(c.lower() for c in word if c.isalnum()))

    if wordIsPalindrome:
        print(f"{word} is a palindrome!")
    else:
        print(f"{word} is not a palindrome.")
